File: OneDrive/Documents/PhaseGateOne/Python/menstrual_calculator.py
from datetime import date
todaysDate = date.today()

def nextFlow(day_of_flow, flow_month, flow_length):
	#To calculate the next menstrual flow.
	
	flow_date = date()

	print(todaysDate)
	
	return todaysDate



flow_month = int(input("Enter month of previous flow: "))
day_of_flow = int(input("Enter day of previous flow: "))
flow_length = int(input("Enter length previous flow: "))
cycle_length = int(input("Enter length menstrual cycle: "))
print(nextFlow(prevDay, prevMonth))

File: OneDrive/Desktop/PhaseGateOne/JavaScript/MenstrualCalculator.js
const prompt = require ("prompt-sync")()
const moment = require ('moment');

function nextFlowDate(monthOfFlow, dayOfFlow, flowLength, nextFlow){
	const date = new Date();

	let year = date.getFullYear();
	let monthLength = new Date(year, monthOfFlow + 1, 0).getDate;
	let month = date.getMonth();
	let dayOfMonth = date.getDay();
	let remainingDaysInMonth = monthLength - dayOfFlow;

	if (nextFlow > remainingDaysInMonth){
		nextFlow -= remainingDaysInMonth; monthOfFlow += 1;
	} else{
		nextFlow += dayOfFlow;
		}
	menstDate = Date(year, monthOfFlow, nextFlow);
	console.log("Next menstruation is", monthLength.toLocaleDateString())
	return monthLength;
};

function ovulationPeriod(monthOfFlow, dayOfFlow){
	const menstDate = new Date(2025, monthOfFlow, dayOfFlow);
	let ovulation = dayOfFlow + 14;
	let daysLeft = menstDate.getMonth().length - dayOfFlow;
	if (ovulation > daysLeft){
		ovulation = 14 - daysLeft;
		monthOfFlow += 1;
		console.log("Ovulation is due on ", ovulation , " of ", monthOfFlow)
		}
	return ovulation;
};


let monthOfFlow = prompt("Enter month of last flow: ")
let dayOfFlow = prompt("Enter day of last flow: ")
let flowLength = prompt("Enter length of last flow: ")
let nextFlow = prompt("Enter of days until next flow: ")

console.log(ovulationPeriod(monthOfFlow, dayOfFlow))
console.log(nextFlowDate(monthOfFlow, dayOfFlow, flowLength, nextFlow))


